hand_eye_calibration/python/hand_eye_calibration/time_alignment.py

```diff
@@ -238,8 +240,9 @@ def compute_aligned_poses(time_stamped_poses_A,
   interval = end_time - start_time
 
   # Resample at the lower frequency to prevent introducing more noise.
-  dt_A = np.mean(np.diff(time_stamped_poses_A_shifted[:, 0]))
-  dt_B = np.mean(np.diff(time_stamped_poses_B[:, 0]))
+  max_diff_s = 1.
```

Comment:
This value is of course a bit arbitrary. What about using a quantiles instead. For instance you could take the 90% smaller values (time diffs) and take their mean. I.e. you could sort the values and then use the ones below the index floor(90% * N) with N being the number of values.

Reply:
yes, fair point! I was actually thinking about just replacing the mean with median, instead of the filtering approach. That should also work rather robustly, no?

Reply:
I also thought about that. It should work as well, but you get a bit less the average delta in the sense, that if there are no big gaps then #(measurements in A - 1) * dt_A ~= duration of A and I'm not so sure how much the approach below kind of builds on this property. But probably for the sake of simplicity a mean is the better choice. The big trouble here is that we simply don't have test cases on which we could validate that we don't make things worse for many cases..